v3_async_wip/v3_async_wip/mqtt_client.py

```diff
@@ -121,13 +123,22 @@ def __init__(
         # Synchronization
         self.connected_cond = asyncio.Condition()
         self.disconnected_cond = asyncio.Condition()
-        self._connection_lock = ConnectionLock()
+        self._connection_lock = asyncio.Lock()
+        self._mid_tracker_lock = asyncio.Lock()
 
-        # Tasks
+        # Tasks/Futures
+        self._network_loop = None
         self._reconnect_daemon = None
+        # NOTE: pending connect is protected by the connection lock
+        # Other pending ops are protected by the _mid_tracker_lock
+        self._pending_connect = None
+        self._pending_subs = {}  # Map mid -> Future
+        self._pending_unsubs = {}  # Map mid -> Future
+        self._pending_pubs = {}  # Map mid -> Future
 
-        # Event Loop
-        self._loop = asyncio.get_running_loop()
+        # Incoming Data
+        self._incoming_messages = asyncio.Queue()
+        self._incoming_filtered_messages = {}  # Map topic -> asyncio.Queue
```

Comment:
> # Map topic -> asyncio.Queue [](http://example.com/codeflow?start=46&length=29)

We should consider using typings -- I think there are ways to slowly add them to an existing codebase and this seems like the right time to start adding them.

Reply:
I'll put a separate PR for this out tomorrow